packages/functional-tests/tests/signin/fxDesktopV3ForceAuth.spec.ts

```diff
@@ -0,0 +1,163 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { test, expect } from '../../lib/fixtures/standard';
+
+const makeUid = () =>
+  [...Array(32)]
+    .map(() => Math.floor(Math.random() * 16).toString(16))
+    .join('');
+
+test.describe('Desktop Sync V3 force auth', () => {
+  test('sync v3 with a registered email, no uid', async ({
+    credentials,
+    pages: {
+      fxDesktopV3ForceAuth,
+      login,
+      signInTokenCode,
+      connectAnotherDevice,
+    },
+  }) => {
+    await fxDesktopV3ForceAuth.openWithReplacementParams(credentials, {
+      uid: undefined,
+    });
+    await login.setPassword(credentials.password);
+    await login.submit();
+    await signInTokenCode.isVisible();
+    await fxDesktopV3ForceAuth.checkWebChannelMessage(
+      'fxaccounts:can_link_account'
+    );
+    await login.fillOutSignInCode(credentials.email);
+    await connectAnotherDevice.isVisible();
+    await fxDesktopV3ForceAuth.checkWebChannelMessage('fxaccounts:login');
```

Comment:
Should this assert true?

Reply:
The exception thrown after X seconds of checking for the message will fail the test.